lib/RWMutex.ts

```diff
@@ -65,10 +76,17 @@ export default class RWMutex {
           },
         });
         if (result.matchedCount > 0) {
+          if (typeof cb === "function") {
+            cb();
+          }
           return;
         }
       } catch (err) {
-        throw new Error(`error aquiring lock ${this._lockID}: ${err.message}`);
+        const formattedError = new Error(`error aquiring lock ${this._lockID}: ${err.message}`);
+        if (typeof cb === "function") {
+          cb(formattedError);
```

Comment:
I actually don't think it's necessary. My understanding is that the `throw` will be captured as a rejected promise because these functions are all tagged `async`. So if you call `lock.lock((err) => {...})`, you'll just be ignoring the Promise returned by the function and the `throw` is essentially a noop in that case.

I can add it if we think it's clearer though.